service/federations/requests.go

```diff
@@ -0,0 +1,206 @@
+	}
+
+	response, err := s.baseClient.DoRequest(ctx, client.DoRequestInput{
+		Body:   nil,
+		Method: http.MethodGet,
+		Path:   path,
+	})
+	if err != nil {
+		//nolint:wrapcheck // DoRequest already wraps the error.
+		return nil, err
+	}
+
+	var federations ListResponse
+	err = client.UnmarshalJSON(response, &federations)
+	if err != nil {
+		return nil, iamerrors.Error{Err: iamerrors.ErrInternalAppError, Desc: err.Error()}
+	}
+	return &federations, nil
+}
+
+// Get returns an info of Federation with federationID.
+func (s *Service) Get(ctx context.Context, federationID string) (*GetResponse, error) {
+	if federationID == "" {
+		return nil, iamerrors.Error{Err: iamerrors.ErrFederationIDRequired, Desc: "No federationID was provided."}
+	}
+
+	path, err := url.JoinPath(apiVersion, "federations", "saml", federationID)
+	if err != nil {
+		return nil, iamerrors.Error{Err: iamerrors.ErrInternalAppError, Desc: err.Error()}
+	}
+
+	response, err := s.baseClient.DoRequest(ctx, client.DoRequestInput{
+		Body:   nil,
+		Method: http.MethodGet,
+		Path:   path,
+	})
+	if err != nil {
+		//nolint:wrapcheck // DoRequest already wraps the error.
+		return nil, err
+	}
+
+	var federation GetResponse
+	err = client.UnmarshalJSON(response, &federation)
+	if err != nil {
+		return nil, iamerrors.Error{Err: iamerrors.ErrInternalAppError, Desc: err.Error()}
+	}
+	return &federation, nil
+}
+
+// Create creates a new Federation.
+func (s *Service) Create(ctx context.Context, input CreateRequest) (*CreateResponse, error) {
+	if input.Name == "" {
+		return nil, iamerrors.Error{
+			Err:  iamerrors.ErrFederationNameRequired,
+			Desc: "No Name for Federation was provided.",
+		}
+	}
+	if input.Issuer == "" {
+		return nil, iamerrors.Error{
+			Err:  iamerrors.ErrFederationIssuerRequired,
+			Desc: "No Issuer for Federation was provided.",
+		}
+	}
+	if input.SSOUrl == "" {
+		return nil, iamerrors.Error{
+			Err:  iamerrors.ErrFederationSSOURLRequired,
+			Desc: "No SSO URL for Federation was provided.",
+		}
+	}
+
+	path, err := url.JoinPath(apiVersion, "federations", "saml")
+	if err != nil {
+		return nil, iamerrors.Error{Err: iamerrors.ErrInternalAppError, Desc: err.Error()}
+	}
+
+	body, err := json.Marshal(input)
+	if err != nil {
+		return nil, iamerrors.Error{Err: iamerrors.ErrInternalAppError, Desc: err.Error()}
+	}
+
+	response, err := s.baseClient.DoRequest(ctx, client.DoRequestInput{
+		Body:   bytes.NewReader(body),
+		Method: http.MethodPost,
+		Path:   path,
+	})
+	if err != nil {
+		//nolint:wrapcheck // DoRequest already wraps the error.
+		return nil, err
+	}
+
+	var federation CreateResponse
+	err = client.UnmarshalJSON(response, &federation)
+	if err != nil {
+		return nil, iamerrors.Error{Err: iamerrors.ErrInternalAppError, Desc: err.Error()}
+	}
+	return &federation, nil
+}
+
+// Check checks if Federation with federationID exists.
+func (s *Service) Check(ctx context.Context, federationID string) error {
```

Comment:
I'm not sure if this method is needed in the client. It is used for better error handling on the frontend.

Reply:
Ok then, removed it from sdk